packages/trip-form/src/TripOptions/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { ReactElement, useState } from "react";
+
+import GeneralSettingsPanel from "../GeneralSettingsPanel";
+
+import FeaturedOption from "./FeaturedOption";
+import FeaturedOptionOverlay from "./FeaturedOptionOverlay";
+import ModeRow from "./ModeRow";
+import TransitOptions from "./TransitOptions";
+import { Company, QueryProps } from "./types";
+import * as S from "./styled";
+
+interface ComponentProps {
+  /**
+   * React element to be rendered below
+   * the rest of the element
+   */
+  footer?: ReactElement;
+
+  /**
+   * List of company objects to include in the
+   * featured options
+   */
+  supportedCompanies: Company[];
+}
+
+type Props = ComponentProps & QueryProps;
+
+/**
+ * This component renders the custom TriMet Mode Selector
+ */
+export default function TripOptions(props: Props): ReactElement {
+  const [featuredOption, setFeaturedOption] = useState(null);
+  const {
+    footer,
+    onQueryParamChange,
+    queryParams,
+    supportedCompanies,
+    supportedModes
```

Comment:
After thinking this through a little more I have a little bit of an adjustment from what is laid out in earlier as to what is going to provide the cleanest interface to supply `TripOptions` with custom icon components that we want to inject given the way those icon exist in trimet.org. This is what I'm thinking:

TripOptions component will get three new props called something like: `SimpleModeIcon`, `DetailedModeIcon`, `CompanyIcon`.  These will be passed in as uninstantiated react components that are wrappers on a set of icons, like the `ModeIcon` component in this project, and each will be passed a single prop that determines which icon is displayed.   SimpleModeIcon will be sent to the `Checkbox` components and be a passed a `mode` prop along the way.  DetailedModeIcon will end in the `TransitOptions` and `FeatureOptions` components and be passed a `mode` prop.  And CompanyIcon will end up in `FeatureOptionOverlay` and be passed a `company` prop.  Am thinking these would all be optional props and fallback to using the image urls if not passed in

@miles-grant-ibigroup how does this sound good to you?

Reply:
yep, good catch, DetailModeIcon should be used there as well

Reply:
added!